### PR TITLE
Fix domain socket default setting in helm

### PR DESCRIPTION
**Pull request by ssz1997**
Solves https://github.com/Alluxio/alluxio/issues/16721

Using persistent volume claim for domain socket doesn't work because:
    1. The access mode has to be `ReadWriteOnce` (can only be mounted for reading/writing on one physical machine) because domain socket is a same-node inter-process communication. It doesn't make sense if the directory exists on a different machine and the communication is through network.
    2. Based on 1 and since worker is Daemonset, each node has only 1 worker, the persistent volume cannot be mounted to multiple workers, and thus all workers but one would not successfully start.
    3. Creating one persistent volume claim for every worker also doesn't work because it's not supported by Daemonset.

Thus we change the default to `hostPath`. K8s will create the dir with root as owner, so we need to do a chmod for alluxio to have proper permission.

### CODE REVIEW

File: integration/kubernetes/helm-chart/alluxio/values.yaml [hunk 1]
```diff
@@ -448,7 +448,7 @@ shortCircuit:
   policy: uuid
   # volumeType controls the type of shortCircuit volume.
   # It can be "persistentVolumeClaim" or "hostPath"
-  volumeType: persistentVolumeClaim
+  volumeType: hostPath
```
Comment (jiacheliu3) [inline integration/kubernetes/helm-chart/alluxio/values.yaml:451]: There are two orthogonal topics and I'd like to discuss separately.

**Topic 1: How to effectively use `hostPath` without too much user trouble**

We've had this struggle on worker tiered storage (that supports hostPath too). Having a initContainer which requests root permission (without users knowing) is somewhat tricky because it brings about a chain of user questions:
1. Why does Alluxio want root permission (and hence rejected by our admin or `PodSecurityPolicy`)?
2. Then what do I do if I want to use Alluxio without root permission? -- We then had to provide the instructions in our docs anyway.

Let's just talk about the default config here, as we want minimal user perception and trouble. We want users to just start Alluxio with a minimal setup and try, and succeed without tweak for 2 hours.

Then supporting `emptyDir` is necessary for such test env, and that doesn't take extra permission mgmt. So do you want the default to be `emptyDir`? My answer would be -- if you look around and see other pods using `emptyDir` as the default volume type, that suggests we guess the user wants the default helm chart to be a test setup. Then use `emptyDir` here as the default, to play along. Otherwise if you feel all other code favor a production/serious deployment like PVC, that we used to assume admins know what they are doing when they see "PVC", play along too.

I've been away from the helm chart for a while and can't remember what we favor here (test or prod). So i'll leave it up to you and your trusted community users for which one to favor -- but favor that consistently.

**Topic 2: Does PVC work at all?**

In short, this domain socket volume is the same as other worker tiered storage volumes. So if we support tiered store to be PVC, that suggests domain socket as PVC works.

You want this volume to be node-local and next to the Pod. Then we use DaemonSet to make sure each worker claims its exclusive local volume. If tiered storage works with PVC, then domain socket would the same way. If the PVC mounts to a PV of `hostPath` or `local`, then each worker will go to its local path even though the PV/PVC types are `ReadWriteOnce`.

File: integration/kubernetes/helm-chart/alluxio/templates/worker/daemonset.yaml [hunk 2]
```diff
@@ -88,6 +88,18 @@ spec:
       {{- if .Values.imagePullSecrets }}
 {{ include "alluxio.imagePullSecrets" . | indent 6 }}
       {{- end}}
+      {{- if and (eq .Values.shortCircuit.enabled true) (eq .Values.shortCircuit.volumeType "hostPath") }}
```
Comment (jiacheliu3) [inline integration/kubernetes/helm-chart/alluxio/templates/worker/daemonset.yaml:91]: One alternative is, instead of initContainer and root perm, would it be better if you need the hostPath to exist (`Directory` instead of `DirectoryOrCreate`). Would a `file not found` be more straightforward to the user than `perm denied`?

**Review by jiacheliu3 — CHANGES_REQUESTED** [review]
Hey thanks for the work! My comment is actually rather high-level. I feel there are two options for the default setup: 
1. A test env where the user just downloads and plays. This default should work with minimal manual work and provide a running but small playground. In this case, every volume should be `emptyDir` to avoid any permission or mapping problem.
2. A relatively serious deployment (potentially prod env). Then we assume the admin reads the doc and knows K8s/alluxio. We give them PVC, asking them to provision PVs manually/automatically in ways they like. The gap is filled with doc and support.

We didn't put `hostPath` as the default in any volume because it is not the best choice in either case. If there are arguments that `hostPath` should be a better default, let's further discuss. 

I feel like the solution to your issue is not immediately obvious, let's discuss offline?

**Comment by ssz1997** [comment]
@jiacheliu3 Agree with if we aim for test env, `emptyDir` would be a better choice than `hostPath`. Let's discuss the PV/PVC offline.

**Comment by ssz1997** [comment]
emptyDir does not show the real path, so it is hard for spark to access it, and thus don't use emptyDir.

We need to pre setup all local PVs before the pvc can be used.
